cli/src/main/java/com/devonfw/tools/ide/tool/graalvm/GraalVm.java

```diff
@@ -0,0 +1,66 @@
+package com.devonfw.tools.ide.tool.graalvm;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link LocalToolCommandlet} for <a href="https://www.graalvm.org/">GraalVM</a>, an advanced JDK with ahead-of-time Native Image compilation.
+ */
+public class GraalVm extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public GraalVm(IdeContext context) {
+
+    super(context, "graalvm", Set.of(Tag.JAVA));
+  }
+
+  @Override
+  public Path getToolPath() {
+
+    return this.context.getSoftwarePath().resolve("extra").resolve(getName());
+  }
+
+  @Override
+  public VersionIdentifier getInstalledVersion() {
+
+    return super.getInstalledVersion(getToolPath());
+  }
+
+  @Override
+  public void run() {
+
+    Path toolPath = getToolPath();
+    try {
+      if (!Files.exists(toolPath)) {
+        Files.createDirectories(toolPath);
+
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to create new directory:" + e);
+    }
+    super.install(true);
+
+  }
+
+  @Override
+  public void postInstall() {
+
+    EnvironmentVariables envVars = this.context.getVariables().getByType(EnvironmentVariablesType.CONF);
+    envVars.set("GRAALVM_HOME", getToolPath().toString(), true);
+    envVars.save();
+    super.postInstall();
+  }

```

Comment:
This is how we solved such problems in the bash code of devonfw-ide.
In IDEasy we have a much smarter solution by simply defining `GRAALVM_HOME` in `IdeVariables`.
WDYT?
p.s.: This is nice to have. We can also merge and later add an improvement for this. Besides there are already other occurrences of this "pattern" already in IDEasy so we could better create a new issue to do a cleanup/refactoring of this pattern with `IdeVariables` for all occurrences and resolve this conversation. Also this is something to be announced and shared in the team so every developer gets aware of this learning.